### ISSUE TITLE
xadd not found

### ISSUE THREAD

**Issue by mankins** [upstash/upstash-redis#349]
Hi. I was [reading about streams on your blog](https://upstash.com/blog/redis-streams-beyond-memory) so naturally wanted to try it out.

I believe that this library doesn't yet have support for streams / `xadd`, is that correct? I'm getting this:

```
redis.xadd is not a function
```

But it's possible I've missed something?

**Comment by chronark** [comment]
streams are not yet added to the sdk.
they work with standard clients or over the rest api though

**Comment by mankins** [comment]
I was able to add this to use REST for now. In case it's helpful:

```
	// see if our redis client has support for xadd, if not add it
	if (!redisClient.xadd) {
		redisClient.xadd = async (key, id, ...args) => {
			const logArgs = args.map((arg) => {
				if (typeof arg === 'object') {
					return JSON.stringify(arg);
				}
				return encodeURIComponent(arg);
			});

			// join the args with a /
			const parts = logArgs.join('/');
			const request = await fetch(`${AUTH_UPSTASH_REDIS_URL}/xadd/${encodeURIComponent(key)}/${encodeURIComponent(id)}/${parts}`, {
				headers: {
					Authorization: `Bearer ${AUTH_UPSTASH_REDIS_TOKEN}`
				}
			});
			const result = await request.json();
			// console.log({result});
			return result;
		};

	}

```

**Comment by ogzhanolguncu** [comment]
XADD and XRANGE are both added in https://github.com/upstash/upstash-redis/releases/tag/v1.23.0.

**Comment by adriancooney** [comment]
@ogzhanolguncu `XREVRANGE` is still not supported - is it possible to get support for it too?

**Comment by ogzhanolguncu** [comment]
@adriancooney Sure I'll add it today.

**Comment by ogzhanolguncu** [comment]
@adriancooney  `XREVRANGE` added along with `XDEL`, `XTRIM` AND `XLEN` in https://github.com/upstash/upstash-redis/releases/tag/v1.26.0

**Comment by adriancooney** [comment]
Thanks @ogzhanolguncu!

**Comment by basst85** [comment]
Thanks a lot @ogzhanolguncu ! 👍Is XREAD also gonna be supported in the near future?

**Comment by ogzhanolguncu** [comment]
@basst85 Probably next week i'll add rest of them to the SDK.

**Comment by ogzhanolguncu** [comment]
@basst85 @haydn Rest of the stream commands added in https://github.com/upstash/upstash-redis/pull/809 and they are available in version 1.28.0. Sorry for being a little later than expected. Have fun with streams 😄.
